idseq/uploader.py

```diff
@@ -35,19 +37,31 @@ def source_type(self):
     def parts(self, max_part_size):
         # Check if any file is over max_part_size and, if so, chunk
         if self.source_type() == 'local' and os.path.getsize(
-                self.path) > max_part_size * 1048576:
+                self.path) > max_part_size:
             part_prefix = self.path + PART_SUFFIX
-            print("splitting large file into {} MB chunks...".format(
-                max_part_size))
-            subprocess.check_output(
-                "split -b {}m {} {}".format(max_part_size, self.path,
-                                            part_prefix),
-                shell=True)
-            return subprocess.check_output(
-                "ls {}*".format(part_prefix), shell=True).decode("utf-8").splitlines()
+            return self.split_file(max_part_size, part_prefix)
         else:
             return [self.path]
 
+    def split_file(self, max_part_size, prefix):
+        print("Splitting large file into {} MB chunks...".format(int(max_part_size // 1E6)))
+        if not os.path.isfile(self.path):
+            print("Sample file not found: {}".format(self.path))
+            return []
+
+        partial_files = []
+        suffix_iter = product(ascii_lowercase, repeat=2)
+        try:
+            with open(self.path, 'rb') as fread:
+                for chunk in iter(lambda: fread.read(max_part_size), b''):
+                    partial_files.append("{}{}".format(prefix, ''.join(next(suffix_iter))))
+                    with open(partial_files[-1], 'bw') as fwrite:
+                        fwrite.write(chunk)
+            return partial_files
+        except StopIteration:
+            print("[ERROR] File too large")
```

Comment:
should we actually re-raise an exception here?

Reply:
No, because it is a client and a well know exception. Should terminate with user friendly error.